api/types/saml_idp_service_provider.go

```diff
@@ -139,3 +140,11 @@ func (s SAMLIdPServiceProviders) Less(i, j int) bool { return s[i].GetName() < s
 
 // Swap swaps two service providers.
 func (s SAMLIdPServiceProviders) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// samlIdPSPEntityDescriptor is a struct used to parse out the entity ID from the service provider
+// XML.
+//
+//nolint:revive // Because we want this to be IdP.
+type samlIdPSPEntityDescriptor struct {
```

Comment:
suggestion: since this type is purely for extracting a single field from some XML and is not relevant at a larger scope, i'd use an anonymous struct at the point it is used:
```
                // Extract just the entityID attribute from the descriptor
                ed := struct {
                        EntityID string `xml:"entityID,attr"`
                }{}
                err := xml.Unmarshal([]byte(s.Spec.EntityDescriptor), &ed)
...
```